lib/repo/mapping.go

```diff
@@ -0,0 +1,150 @@
+package repo
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+const missingRepo = "Device repo is missing"
+
+var path string
+
+func init() {
+	//p, err := os.Executable()
+	//if err != nil {
+	//	logrus.Error(err)
+	//}
+
+	path = filepath.Join(baseDir, file)
+}
+
+// DeviceImage contains url, title, username and password which are used after flashing
+type DeviceImage struct {
+	Url   string `json`
+	Title string `json`
+	User  string `json`
+	Pass  string `json`
+}
+
+// DeviceMapping is a collection of device, it sub-types and sets of images for these devices
+type DeviceMapping struct {
+	Name   string           `json`
+	Sub    []*DeviceMapping `json`
+	Images []DeviceImage    `json`
+
+	dir string
+	Url DeviceImage
```

Comment:
__[golint]__
* [golint] struct field Url should be URL

<!-- Comment by [SideCI](https://sideci.com) -->